core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberAccessorImplementor.java

```diff
@@ -78,35 +79,55 @@ public String getName() {
-     * @param annotationClass The annotation it was annotated with (used for
-     *        error reporting)
+     * @param className never null
+     * @param classOutput never null, defines how to write the bytecode
+     * @param memberInfo never null, member to generate MemberAccessor methods implementation for
      */
-    public static void defineAccessorFor(ClassCreator classCreator, GizmoMemberDescriptor member,
-            Class<? extends Annotation> annotationClass) {
-        classCreator.getFieldCreator("genericType", Type.class)
-                .setModifiers(Modifier.FINAL);
-        classCreator.getFieldCreator("annotatedElement", AnnotatedElement.class)
-                .setModifiers(Modifier.FINAL);
-
-        GizmoMemberInfo memberInfo = new GizmoMemberInfo(member, annotationClass);
-        // ************************************************************************
-        // MemberAccessor methods
-        // ************************************************************************
-        createConstructor(classCreator, memberInfo);
-        createGetDeclaringClass(classCreator, memberInfo);
-        createGetType(classCreator, memberInfo);
-        createGetGenericType(classCreator);
-        createGetName(classCreator, memberInfo);
-        createGetSpeedNote(classCreator);
-        createSupportSetter(classCreator, memberInfo);
-        createExecuteGetter(classCreator, memberInfo);
-        createExecuteSetter(classCreator, memberInfo);
-        createGetAnnotation(classCreator);
+    public static void defineAccessorFor(String className, ClassOutput classOutput, GizmoMemberInfo memberInfo) {
+        Class<? extends AbstractGizmoMemberAccessor> superClass = getCorrectSuperclass(memberInfo);
+        try (ClassCreator classCreator = ClassCreator.builder()
+                .className(className)
+                .superClass(superClass)
+                .classOutput(classOutput)
+                .setFinal(true)
+                .build()) {
+            classCreator.getFieldCreator("genericType", Type.class)
+                    .setModifiers(Modifier.FINAL);
+            classCreator.getFieldCreator("annotatedElement", AnnotatedElement.class)
+                    .setModifiers(Modifier.FINAL);
+
+            // ************************************************************************
+            // MemberAccessor methods
+            // ************************************************************************
+            createConstructor(classCreator, memberInfo);
+            createGetDeclaringClass(classCreator, memberInfo);
+            createGetType(classCreator, memberInfo);
+            createGetGenericType(classCreator);
+            createGetName(classCreator, memberInfo);
+            createExecuteGetter(classCreator, memberInfo);
+            if (superClass == AbstractReadWriteGizmoMemberAccessor.class) {
+                createExecuteSetter(classCreator, memberInfo);
```

Comment:
Not 100 % sure about the implementation of this method in this new context. Can you please double-check, @Christopher-Chianelli ?

Reply:
Looks correct in the new context (methods are only generated when they are not implemented by an abstract base class); I am guessing you change the signature to take ClassOutput and modified the OptaPlanner and Quarkus call site to pass it.